h2o-algos/src/main/java/hex/tree/SharedTreeMojoWriter.java

```diff
@@ -38,6 +38,28 @@ protected void writeModelData() throws IOException {
       writekv("calib_method", "platt");
       writekv("calib_glm_beta", beta);
     }
+    writekv("_genmodel_encoding", model.getGenModelEncoding());
+    String[] origNames = model._output._origNames;
+    if (origNames != null) {
```

Comment:
loot at writeDomains() in AbstractMojoWriter, I think we should handle this in a simmilar manner, same for reading

Reply:
fixed